### ISSUE TITLE
Support for azurerm_marketplace_agreement data source

### ISSUE THREAD

**Issue by ghost** [hashicorp/terraform-provider-azurerm#17664]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

It would be good to have a data source to retrieve the current marketplace agreement status on an image. I have code that affects the same subscription but using different states. The second deployment fails because the marketplace terms for an image have already been agreed.

### New or Affected Resource(s)/Data Source(s)

azurerm_marketplace_agreement

### Potential Terraform Configuration

```hcl
resource "azurerm_marketplace_agreement" "dmvpn" {
  count = var.primary_region ? 1 : 0
  publisher = var.dmvpn_vm_publisher
  offer     = var.dmvpn_vm_offer
  plan      = var.dmvpn_vm_sku
}

data "azurerm_marketplace_agreement" "dmvpn" {
  count = var.primary_region ? 0 : 1
  publisher = var.dmvpn_vm_publisher
  offer     = var.dmvpn_vm_offer
  plan      = var.dmvpn_vm_sku
}

locals {
  dmvpn_marketplace_agreement = var.primary_region ? azurerm_marketplace_agreement.dmvpn : data.azurerm_marketplace_agreement.dmvpn
}

resource "azurerm_linux_virtual_machine" "dmvpn_vm" {
  depends_on = [local.dmvpn_marketplace_agreement]
  [...]
}
```

**Comment by myc2h6o** [comment]
I found this already supported by #19628

**Comment by tombuildsstuff** [comment]
Fixed via #19628, thanks @myc2h6o
